src/key_info_managers/mod.rs

```diff
@@ -261,24 +414,25 @@ impl KeyInfoManagerClient {
         }
     }
 
-    /// Returns a Vec of ApplicationName of clients having keys in the provider.
+    /// Returns a Vec of ApplicationIdentity of clients having keys in the provider.
     ///
     /// # Errors
     ///
     /// Returns an error as a String if there was a problem accessing the Key Info Manager.
-    pub fn list_clients(&self) -> parsec_interface::requests::Result<Vec<ApplicationName>> {
+    pub fn list_clients(&self) -> parsec_interface::requests::Result<Vec<ApplicationIdentity>> {
         let key_info_manager_impl = self
             .key_info_manager_impl
             .read()
             .expect("Key Info Manager lock poisoned");
-        let key_triples = key_info_manager_impl
-            .get_all(self.provider_id)
+        let key_identities = key_info_manager_impl
+            .get_all(self.provider_identity.clone())
             .map_err(to_response_status)?;
-        let mut clients = Vec::new();
 
-        for key_triple in key_triples {
-            if !clients.contains(key_triple.app_name()) {
-                let _ = clients.push(key_triple.app_name().clone());
+        let mut clients = Vec::new();
+        for key_identity in key_identities {
+            // TODO: Check this comparison
+            if !clients.contains(&key_identity.application) {
+                let _ = clients.push(key_identity.application.clone());
             }
         }
```

Comment:
Could do the filtering as a `HashSet`, but don't know if it's that helpful since you'd have to convert to `Vec` afterwards.